runtime/src/iree/hal/drivers/vulkan/vulkan_device.cc

```diff
@@ -1712,8 +1712,13 @@ static iree_status_t iree_hal_vulkan_device_queue_execute(
   };
   IREE_RETURN_IF_ERROR(queue->Submit(1, &batch));
   // HACK: we don't track async resource lifetimes so we have to block.
-  return iree_hal_semaphore_list_wait(signal_semaphore_list,
-                                      iree_infinite_timeout());
+  IREE_RETURN_IF_ERROR(iree_hal_semaphore_list_wait(signal_semaphore_list,
```

Comment:
I should add a protection using `#if IREE_TRACING_FEATURES & IREE_TRACING_FEATURE_INSTRUMENTATION_DEVICE` for this.